azurerm/resource_arm_cognitive_account.go

```diff
@@ -0,0 +1,258 @@
+package azurerm
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/cognitiveservices/mgmt/2017-04-18/cognitiveservices"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmCognitiveAccount() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmCognitiveAccountCreate,
+		Read:   resourceArmCognitiveAccountRead,
+		Update: resourceArmCognitiveAccountUpdate,
+		Delete: resourceArmCognitiveAccountDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"kind": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateCognitiveAccountKind,
+			},
+
+			"sku": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     false,
+							ValidateFunc: validateCognitiveAccountSkuName,
+						},
+
+						"tier": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     false,
+							ValidateFunc: validateCognitiveAccountSkuTier,
```

Comment:
can we remove the validation for the SKU Name & Tier? In general it's simpler to let Azure handle that as it can vary by account/region